packages/buidler-core/src/builtin-tasks/compile.ts

```diff
@@ -159,229 +168,338 @@ export default function () {
-      log(`'${groupsFilteredOutCount}' groups were filtered out`);
+        const neededCompilationJobs = compilationJobs.filter((job) =>
+          needsCompilation(job, solidityFilesCache)
+        );
 
-      return emittingCompilationGroups;
-    }
-  );
+        const jobsFilteredOutCount =
+          neededCompilationJobs.length - compilationJobs.length;
+        log(`'${jobsFilteredOutCount}' jobs were filtered out`);
+
+        return neededCompilationJobs;
+      }
+    );
 
   /**
-   * Receives a list of compilation groups and returns a new list where some of
-   * the groups might've been merged.
+   * Receives a list of compilation jobs and returns a new list where some of
+   * the jobs might've been merged.
    */
-  internalTask(
-    TASK_COMPILE_MERGE_COMPILATION_GROUPS,
-    async ({
-      compilationGroups,
-    }: {
-      compilationGroups: CompilationGroup[];
-    }): Promise<CompilationGroup[]> => {
-      return mergeCompilationGroupsWithoutBug(compilationGroups);
-    }
-  );
+  internalTask(TASK_COMPILE_SOLIDITY_MERGE_COMPILATION_JOBS)
+    .addParam("compilationJobs", undefined, undefined, types.any)
+    .setAction(
+      async ({
+        compilationJobs,
+      }: {
+        compilationJobs: CompilationJob[];
+      }): Promise<CompilationJob[]> => {
+        return mergeCompilationJobsWithoutBug(compilationJobs);
+      }
+    );
 
   /**
    * Prints a message when there's nothing to compile.
    */
-  internalTask(TASK_COMPILE_LOG_NOTHING_TO_COMPILE, async () => {
-    console.log("Nothing to compile");
-  });
+  internalTask(TASK_COMPILE_SOLIDITY_LOG_NOTHING_TO_COMPILE)
+    .addParam("quiet", undefined, undefined, types.boolean)
+    .setAction(async ({ quiet }: { quiet: boolean }) => {
+      if (!quiet) {
+        console.log("Nothing to compile");
+      }
+    });
 
   /**
-   * Receives a list of compilation groups and sends each one to be compiled.
+   * Receives a list of compilation jobs and sends each one to be compiled.
    */
-  internalTask(
-    TASK_COMPILE_COMPILE_GROUPS,
-    async (
-      {
-        compilationGroups,
-        solidityFilesCache,
-        force,
-      }: {
-        compilationGroups: CompilationGroup[];
-        solidityFilesCache: SolidityFilesCache;
-        force: boolean;
-      },
-      { run }
-    ) => {
-      if (compilationGroups.length === 0) {
-        log(`No compilation groups to compile`);
-        await run(TASK_COMPILE_LOG_NOTHING_TO_COMPILE);
-        return;
-      }
+  internalTask(TASK_COMPILE_SOLIDITY_COMPILE_JOBS)
+    .addParam("compilationJobs", undefined, undefined, types.any)
+    .addParam("quiet", undefined, undefined, types.boolean)
+    .setAction(
+      async (
+        {
+          compilationJobs,
+          quiet,
+        }: {
+          compilationJobs: CompilationJob[];
+          quiet: boolean;
+        },
+        { run }
+      ): Promise<{ emittedArtifactsPerJob: EmittedArtifactsPerJob }> => {
```

Comment:
Why not just `EmittedArtifactsPerJob`?

Reply:
Which could be a `Map<CompilationJob, ...>`, right?

Reply:
Also, `ArtifactsEmittedPerJob` sounds better than `EmittedArtifactsPerJob`.

Reply:
> Why not just EmittedArtifactsPerJob?

I used an object so that we can add more things to the interface in a backwards compatible way.

> I'm guessing you didn't use an object because of the keys being objects.

Right.

> Also, ArtifactsEmittedPerJob sounds better than EmittedArtifactsPerJob.

Agree, will rename it.

Reply:
> I used an object so that we can add more things to the interface in a backwards compatible way.

Makes sense